src/gui/symbology/qgscategorizedsymbolrendererwidget.cpp

```diff
@@ -535,8 +535,11 @@ QgsCategorizedSymbolRendererWidget::QgsCategorizedSymbolRendererWidget( QgsVecto
   }
 
   mCategorizedSymbol.reset( QgsSymbol::defaultSymbol( mLayer->geometryType() ) );
-  btnChangeCategorizedSymbol->setSymbolType( mCategorizedSymbol->type() );
-  btnChangeCategorizedSymbol->setSymbol( mCategorizedSymbol->clone() );
+  if ( mCategorizedSymbol )
+  {
+    btnChangeCategorizedSymbol->setSymbolType( mCategorizedSymbol->type() );
```

Comment:
Hmmmm, we'll still need to call this method somehow. The layer itself will usually have a wkb type available even when it's invalid. Can you use this to set the symbol type when mCategorizedSymbol doesn't exist?

Reply:
The symbol is created from the layer type, just one line before and the type is `UnknownGeometry` https://github.com/qgis/QGIS/pull/32043/files/5b12071026be7741e2426124624217cc136572c3#diff-bcfbc8c19bf3c403c220c39a7756db57R537

Reply:
Ah, missed that. So the crash must only be happening with an older project, before we started recording the wkb type alongside layers in projects....

Reply:
Anyway, +1 to merge. does graduated need the same fix?

Reply:
Yes, it's an older project.
And yea, apparently graduated too

Reply:
Just fixed whatever crash I could trigger.